src/DeploymentProvider/DeploymentHelper.ts

```diff
@@ -422,4 +418,55 @@ export class DeploymentHelper {
         const data = fs.readFileSync(file);
         return JSON.parse(data.toString());
     }
+
+    private static async printLatestAppInstanceLog(client: asa.AppPlatformManagementClient, params: ActionParameters) {
+        const logStream = await this.logStreamConstructor(client, params);
+        const deploymentResource = await client.deployments.get(params.resourceGroupName, params.serviceName, params.appName, params.deploymentName);
+        const instances = deploymentResource.properties.instances;
+        let startTime = instances[0].startTime;
+        let instanceName = instances[0].name;
+
+        // print the newly created instance log
+        for (const tempInstance of instances) {
+            if (tempInstance.startTime > startTime) {
+                startTime = tempInstance.startTime;
+                instanceName = tempInstance.name;
+            }
+        }
+        let streamingUrl = `https://${logStream["baseUrl"]}/api/logstream/apps/${params.appName}/instances/${instanceName}`;
+        const logParams: any = {};
+        logParams['tailLines'] = 500;
+        logParams['limitBytes'] = 1024 * 1024;
+        logParams['sinceSeconds'] = 300;
+        logParams['follow'] = true;
+        const credentials = Buffer.from(`primary:${logStream["primaryKey"]}`).toString('base64');
+        const auth = { "Authorization" : `Basic ${credentials}` };
+        var url = new URL(streamingUrl)
+        url.search = new URLSearchParams(logParams).toString();
+        fetch(url, {method: 'GET', headers : auth })
+            .then(response => {
+            if (response.ok) {
+              const stream = response.body;
+              // Pipe the stream to stderr
+              stream.pipe(process.stderr);
+            } else {
+              // Handle error response
+              console.error('Error:', response.status, response.statusText);
+            }
+          })
+          .catch(error => {
+            // Handle network error
+            console.error('Error:', error.message);
+          });;
+    }
+
+    private static async deployWithLog(client: asa.AppPlatformManagementClient, params: ActionParameters, deploymentResource: asa.DeploymentResource) {
+        try {
+            const response = await client.deployments.beginCreateOrUpdateAndWait(params.resourceGroupName, params.serviceName, params.appName, params.deploymentName, deploymentResource);
+            core.debug('deploy response: ' + JSON.stringify(response));
+        } catch (e:any) {
+            await this.printLatestAppInstanceLog(client, params);
```

Comment:
I suggest we first output some hint message that we met some exception (maybe log the exception first?), and then print some message that we will try to stream the latest app logs to the console. Then start dumping the logs.

What's the behavior in CLI?

Reply:
CLI will print app logs and throw the error at the end. 
![image](https://github.com/Azure/spring-apps-deploy/assets/86712051/89b473e3-4d55-4cbc-ac48-b293663ba222)

I add one hint message for users to check the app log.
![image](https://github.com/Azure/spring-apps-deploy/assets/86712051/2eea35d4-d6b2-45af-ae81-5c41c9cc5a97)